pkg/cmd/deadman/operator_staleness_check.go

```diff
@@ -0,0 +1,177 @@
+package deadman
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	configv1 "github.com/openshift/api/config/v1"
+	configeversionedclient "github.com/openshift/client-go/config/clientset/versioned"
+	configv1versionedtypes "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+	configv1informers "github.com/openshift/client-go/config/informers/externalversions"
+	configv1lister "github.com/openshift/client-go/config/listers/config/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+)
+
+type OperatorStalenessChecker struct {
+	// operatorToDeadlineForResponse is a map from operator name to the time at which the condition will be considered stale
+	// and the condition will be reset to unknown.
+	operatorToDeadlineForResponse map[string]time.Time
+
+	durationAllowedForOperatorResponse time.Duration
+
+	clusterOperatorClient configv1versionedtypes.ClusterOperatorInterface
+	clusterOperatorLister configv1lister.ClusterOperatorLister
+
+	queue workqueue.RateLimitingInterface
+}
+
+// NewOperatorStalenessChecker is a controller that sets operator conditions status, reason, and messages when the operator
+// fails to respond to a challenge and correct a message.
+func NewOperatorStalenessChecker(
+	configClient configeversionedclient.Interface,
+	configInformers configv1informers.SharedInformerFactory,
+	durationAllowedForOperatorResponse time.Duration,
+) *OperatorStalenessChecker {
+
+	c := &OperatorStalenessChecker{
+		operatorToDeadlineForResponse:      map[string]time.Time{},
+		durationAllowedForOperatorResponse: durationAllowedForOperatorResponse,
+
+		clusterOperatorClient: configClient.ConfigV1().ClusterOperators(),
+		clusterOperatorLister: configInformers.Config().V1().ClusterOperators().Lister(),
+
+		queue: workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "OperatorStalenessChecker"),
+	}
+
+	configInformers.Config().V1().ClusterOperators().Informer().AddEventHandlerWithResyncPeriod(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				clusterOperator := obj.(*configv1.ClusterOperator)
+				c.queue.Add(clusterOperator.Name)
+			},
+			UpdateFunc: func(oldObj, newObj interface{}) {
+				clusterOperator := newObj.(*configv1.ClusterOperator)
+				c.queue.Add(clusterOperator.Name)
+			},
+		},
+		1*time.Minute,
+	)
+
+	return c
+}
+
+func isCheckingForStaleness(clusterOperator *configv1.ClusterOperator) bool {
+	for _, condition := range clusterOperator.Status.Conditions {
+		if strings.HasPrefix(condition.Message, "Checking for stale status") {
```

Comment:
We can avoid this with explicit field, however I think this is fine for initial implementation :-)